deployments/azure/templates/bicep/all-in-one/all-in-one-with-nlb.bicep

```diff
@@ -0,0 +1,238 @@
+      )
+    ),
+    base64(
+      format(
+        loadTextContent('../../../scripts/jupyter.service'),
+        registry,
+        jupyterRepository,
+        JupyterVersion,
+        JupyterHttpPort,
+        JupyterToken
+      )
+    )
+  )
+)
+
+resource rg 'Microsoft.Resources/resourceGroups@2022-09-01' existing = {
+  name: ResourceGroupName
+}
+
+resource network 'Microsoft.Network/virtualNetworks@2022-11-01' existing = {
+  scope: rg
+  name: Network
+}
+
+resource subnet 'Microsoft.Network/virtualNetworks/subnets@2022-11-01' existing = {
+  parent: network
+  name: Subnet
+}
+
+module vault '../modules/vault/vault.bicep' = if (UseKeyVault == 'New') {
+  scope: rg
+  name: 'vault'
+  params: {
+    encryptVolumes: true
+    keyVaultName: AiUnlimitedName
+    location: rg.location
+    tags: Tags
+  }
+}
+
+module vaultAccessPolicy '../modules/vault/access-policy.bicep' = if (UseKeyVault == 'New') {
+  scope: rg
+  name: 'vault-access-policy'
+  params:{
+    vaultName: AiUnlimitedName
+    accessPolicy:  {
+        tenantId: subscription().tenantId
+        objectId: aiUnlimited.outputs.PrincipleId
+        permissions: {
+          keys: [
+            'Create', 'Delete', 'Get', 'List', 'Update', 'Purge', 'Recover', 'Decrypt', 'Encrypt'
+            'Sign', 'UnwrapKey', 'Verify', 'WrapKey', 'GetRotationPolicy', 'SetRotationPolicy'
+          ]
+          secrets: [ 'Get', 'Set', 'Delete', 'List', 'Purge' ]
+          storage: [ 'Get' ]
+        }
+      }
+  }
+}
+
+module firewall '../modules/firewall.bicep' = {
+  scope: rg
+  name: 'firewall'
+  params: {
+    location: rg.location
+    name: SecurityGroup
+    accessCidrs: AccessCIDRs
+    sshAccess: AllowPublicSSH
+    aiUnlimitedHttpPort: AiUnlimitedHttpPort
+    aiUnlimitedGrpcPort: AiUnlimitedGrpcPort
+    jupyterHttpPort: JupyterHttpPort
+    sourceAppSecGroups: SourceAppSecGroups
+    detinationAppSecGroups: detinationAppSecGroups
+    tags: Tags
+  }
+}
+
+module nlb '../modules/nlb.bicep' = {
+  scope: rg
+  name: 'loadbalancer'
+  params: {
+    name: AiUnlimitedName
+    location: rg.location
+    dnsPrefix: dnsLabelPrefix
+    aiUnlimitedHttpPort: AiUnlimitedHttpPort
+    aiUnlimitedGrpcPort: AiUnlimitedGrpcPort
+    jupyterHttpPort: JupyterHttpPort
+    tags: Tags
+  }
+}
+
+module aiUnlimited '../modules/instance.bicep' = {
+  scope: rg
+  name: 'ai-unlimited'
+  params: {
+    location: rg.location
+    name: AiUnlimitedName
+    adminUsername: 'azureuser'
+    sshPublicKey: PublicKey
+    dnsLabelPrefix: dnsLabelPrefix
+    vmSize: InstanceType
+    subnetId: subnet.id
+    networkSecurityGroupID: firewall.outputs.Id
+    osVersion: OSVersion
+    cloudInitData: cloudInitData
+    usePersistentVolume: UsePersistentVolume
+    persistentVolumeSize: PersistentVolumeSize
+    existingPersistentVolume: ExistingPersistentVolume
+    nlbName: AiUnlimitedName
+    nlbPoolNames: nlb.outputs.nlbPools
+    usePublicIp: AllowPublicSSH
+    tags: Tags
+  }
+}
```

Comment:
I tried deploying an NLB deployment, but am running into public IP address conflicts:
@
<img width="1531" alt="Screenshot 2024-04-15 at 12 56 31 PM" src="https://github.com/Teradata/ai-unlimited/assets/166751279/9d80b76c-f607-4696-b895-1fba64155d12">

Is the nlb-inbound DNS label supposed to be the same as the ai-unlimited instance's DNS label?

Reply:
I was able to deploy NLB deployment with slight changes:
<img width="726" alt="Screenshot 2024-04-16 at 3 29 53 PM" src="https://github.com/Teradata/ai-unlimited/assets/166751279/5fb8aa9e-3588-4882-a3a9-a43f1b462cc4">